packages/flutter/lib/src/gestures/recognizer.dart

```
@@ -202,7 +202,13 @@ abstract class PrimaryPointerGestureRecognizer extends OneSequenceGestureRecogni
         resolve(GestureDisposition.rejected);
         stopTrackingPointer(primaryPointer);
       } else {
-        handlePrimaryPointer(event);
+        try {
+          handlePrimaryPointer(event);
+        } catch (e) {
+          resolve(GestureDisposition.rejected);
+          stopTrackingPointer(primaryPointer);
+          rethrow;
```

maybe instead of rethrowing we should just report the exception?